### DISCUSSION TITLE
Traits Attributes not reflected when importing an exported template

### DISCUSSION THREAD

**Discussion by saharhagekr** [GrapesJS/grapesjs#5658] [Q&A]
quick question
when exporting a template, then importing it again in the editor, i am losing the block attributes

FOR EXAMPLE:

editor.DomComponents.addType('sociallink', {
        model: {
            defaults: {
                name: 'Social link',
                selectable: false,
                stylable: false,
                tagName: 'a',
                hoverable: false,
                highlightable: false,
                draggable: false,
                traits: [
                    {
                        type: 'input',
                        name: 'href'
                    },
                ],
                style: {
                    'margin': '0px !important'
                }
            }
        }
    })

If i have a sociallink block in my template, when i export/import, the draggable, hoverable, highlightable etc.... are not reflected. Even though the data-gjs-type is set to sociallink

**Comment by mosh-tudor** [comment]
@saharhagekr

You need to add `draggable, hoverable, highlightable...` like this (with `data-gjs-`):
```html
<a data-gjs-type="sociallink"
   data-gjs-draggable="false"
   data-gjs-highlightable="false"
   data-gjs-draggable="false"
   href="#">
HELLO!!!!
</a>
```

**Reply by saharhagekr** [reply]
@mosh-tudor thank you so much for the reply. Isn't there a way to add them dynamically from the traits manager? I have a lot of components i can't add them manually to each

**Comment by artf** [comment]
@saharhagekr I hope you're not using HTML/CSS for storing/loading your projects...

That would be horribly wrong https://grapesjs.com/docs/modules/Storage.html#project-data

<img width="872" alt="Screenshot 2024-02-11 at 14 08 58" src="https://github.com/GrapesJS/grapesjs/assets/11614725/9b187603-35a2-45ee-8c52-abe34b306690">
